server/controllers/TodoController.js

```diff
@@ -0,0 +1,154 @@
+                res.status(500).json({ message: "Internal Server Error" });
+            });
+    }
+
+    static async put(req, res) {
+        try {
+            const id = Number(req.params.id);
+            const obj = {
+                title: req.body.title,
+                description: req.body.description,
+                status: req.body.status,
+                due_date: req.body.due_date
+            }
+            const todo = await Todo.update(obj, {
+                where: {
+                    id
+                },
+                returning: true
+            });
+            if (todo[0] === 0) {
+                res.status(404).json({ message: "Data is not found." })
+            } else {
+                res.status(200).json(todo[1][0]);
+            }
+        } catch (err) {
+            if (err.name === "SequelizeValidationError") {
+                let errors = [];
+                let messages = [];
+                for (let i = 0; i < err.errors.length; i++) {
+                    if(!errors.includes(err.errors[i].message)) {
+                        errors.push(err.errors[i].message);
+                        messages.push({ message: err.errors[i].message });
+                    }
+                }
+                res.status(400).json(messages);
+            } else {
+                res.status(500).json({ message: "Internal Server Error" });
+            }
+        }
+    }
+
+    static async patch(req, res) {
+        try {
+            const id = Number(req.params.id);
+            const obj = {
+                status: req.body.status,
+            }
+            const todo = await Todo.update(obj, {
+                where: {
+                    id
+                },
+                returning: true
+            });
+            if (todo[0] === 0) {
+                res.status(404).json({ message: "Data is not found." })
+            } else {
+                res.status(200).json(todo[1][0]);
+            }
+        } catch (err) {
+            if (err.name === "SequelizeValidationError") {
+                let errors = [];
+                let messages = [];
+                for (let i = 0; i < err.errors.length; i++) {
+                    if(!errors.includes(err.errors[i].message)) {
+                        errors.push(err.errors[i].message);
+                        messages.push({ message: err.errors[i].message });
+                    }
+                }
+                res.status(400).json(messages);
+            } else {
+                res.status(500).json({ message: "Internal Server Error" });
+            }
+        }
+    }
+
+    static async delete(req, res) {
+        try {
+            const id = Number(req.params.id);
+            const obj = {
+                title: req.body.title,
+                description: req.body.description,
+                status: req.body.status,
+                due_date: req.body.due_date
+            }
+            const todo = await Todo.destroy({
```

Comment:
penamaan variabel mungkin bisa diganti dengan result kali yah , 
karena hasilnya kan  menggambarkan delete success or not
```suggestion
            const result = await Todo.destroy({
```